RedacteurPortaal.Grains/Grains/ArchiveGrain.cs

```diff
@@ -17,37 +20,203 @@ public class ArchiveGrain : Grain, IArchiveGrain
         this.archive = archive;
     }
 
-    public async Task Delete()
+    public Task<ArchiveModel> Get()
     {
-        await this.archive.ClearStateAsync();
+        var state = this.archive.State;
+        return Task.FromResult(state);
+    }
+
+    public async Task<MediaVideoItem> GetVideoItem(Guid guid)
+    {
+        var videoItems = this.archive.State.MediaVideoItems;
+        var videoItem = videoItems.Single(x => x.Id.Equals(guid));
+        
+        await return Task.FromResult(videoItem)
```

Comment:
je returned hier en daarna voer je nog logica uit, daar komt de code nooit

Reply:
is nu aangepast